docs/customization_guide/inference_protocols.md

```diff
@@ -432,3 +432,165 @@ JavaCPP-presets. You can do this using the following steps:
 ```bash
  $ mvn compile exec:java -Djavacpp.platform=linux-x86_64 -Dexec.args="<your input args>"
 ```
+
+## Kafka Protocol
+
+Triton exposes the Kafka endpoint based on [standard inference
+protocols](https://github.com/kserve/kserve/tree/master/docs/predict-api/v2)
+that have been proposed by the [KServeproject](https://github.com/kserve),
+with some minor adjustments.
+
+In order to start the endpoint, Triton requires the following command line
+arguements to be specified with valid values:
+
+* "--kafka-port" : The port of the kafka broker (e.g. localhost:9092).
+* "--kafka-consumer-topics" : One or several comma-separated topics.
+* "--kafka-producer-topic" : Single, default topic to be used by Triton's producer when "response_topic" is not specified in an "inference_request_header". 
+
+The following sections will outline the Kafka request and response message
+designs. It is intended that the schemas mentioned below be constructed
+using JSON before being converted into a string and sent to the Kafka
+broker. As such, the schemas outlined below will be represented as
+JSON objects for simplicity, where $number and $string refer to the
+fundamental JSON types. #optional indicates an optional JSON field.
+$string($number) refers to an integer value that has been converted
+to a string.
+
+## Kafka Inference Request
+Once the Kafka endpoint is running, Triton will begin consuming inference
+requests from the provided topics. Triton expects inference requests stored
+in the Kafka broker to be comprised of a header and a payload. 
+
+#### Message Header
+The inference request header should have the following format:
+
+```C++
+$inference_request_header
+{
+  "id" : $string($number),
```

Comment:
What does this mean? Are we expecting string or number?

Reply:
I had a line about this above. The goal was to inform the user that though this field should be a number, it will ultimately be converted to a string when the message it passed to Triton. But I'll remove as it is confusing.